pkg/monitor/cluster/machineconfigpool_test.go

```diff
@@ -0,0 +1,72 @@
+package cluster 
+
+//Copyright (c) Microsoft Corporation.
+//Licensed under the Apache License 2.0 
+
+import (
+	"context"
+	"testing"
+	"github.com/golang/mock/gomock"
+	"github.com/openshift/machine-config-operator/pkg/generated/clientset/versioned/fake"
+	v1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	mock_metrics "github.com/Azure/ARO-RP/pkg/util/mocks/metrics"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestEmitMachineConfigPoolMetrics(t *testing.T) {
+	ctx := context.Background()
+
+	mcpclient := fake.NewSimpleClientset(&v1.MachineConfigPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "machine-config-pool",
+		},
+		Status: v1.MachineConfigPoolStatus{
+			Conditions: []v1.MachineConfigPoolCondition{
+				{
+					Type: v1.MachineConfigPoolUpdated,
+					Status: corev1.ConditionFalse,
+				},
+			},
+		},
+	}, &v1.MachineConfigPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "machine-config-pool2",
+		},
+		Status: v1.MachineConfigPoolStatus{
+			Conditions: []v1.MachineConfigPoolCondition{
+				{
+					Type: v1.MachineConfigPoolUpdating,
+					Status: corev1.ConditionTrue,
+				},
+			},
+		},
+	})
+		
+
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	m := mock_metrics.NewMockInterface(controller)
+
+	mon := &Monitor{
+		mcp: mcpclient,
+		m: m,
+	}
+	m.EXPECT().EmitGauge("machineconfigpools.count", int(2), map[string]string{})
+	
+	m.EXPECT().EmitGauge("machineconfigpools.condition.count", int64(1), map[string]string{
+		"machineconfigpool": "machine-config-pool",
+		"condition": "NotUpdated",
+	})
+
+	m.EXPECT().EmitGauge("machineconfigpools.condition.count", int64(1), map[string]string{
+		"machineconfigpool": "machine-config-pool2",
+		"condition": "Updating",
+	})
+
+	err := mon.emitMachineConfigPoolMetrics(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
```

Comment:
please set your editor to append a newline on each file!

Reply:
.editorconfig wasn't present to fix change on save 🤦